### PR TITLE
[WIP] Combining into one ConfirmChangeEmail

### PR DESCRIPTION
**Pull request by ShmuelTreiger**
@aapeliv Do you want something like this?

(Tests aren't passing, just did this quickly)

[Skip CI]

### CODE REVIEW

File: app/backend/src/couchers/servicers/auth.py [hunk 1]
```diff
@@ -424,19 +424,31 @@ def CompletePasswordReset(self, request, context):
...
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.INVALID_TOKEN)
 
-    # Only called if user doesn't have password
-    def ConfirmChangeEmailWithOldAddress(self, request, context):
+    def ConfirmChangeEmail(self, request, context):
         with session_scope() as session:
-            user = (
-                session.query(User)
-                .filter(User.old_email_token == request.change_email_token)
+            query = session.query(User).filter(User.id == context.user_id)
```
Comment (aapeliv) [inline app/backend/src/couchers/servicers/auth.py]: The user should be able to verify the token without logging in (e.g. did something on computer and then clicks on email on the phone).

In general in open servicers (like auth is), we don't have `context.user_id` defined.

File: app/backend/src/couchers/servicers/auth.py [hunk 2]
```diff
@@ -424,19 +424,31 @@ def CompletePasswordReset(self, request, context):
...
+                .filter(User.new_email_token_created <= func.now())
+                .filter(User.new_email_token_expiry >= func.now())
+                .one()
+            ):
+                return self.ConfirmChangeEmailWithNewAddress(request, context)
+            else:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.INVALID_TOKEN)
 
+    # Only called if user doesn't have password
+    def ConfirmChangeEmailWithOldAddress(self, request, context):
```
Comment (aapeliv) [inline app/backend/src/couchers/servicers/auth.py]: You should remove this, right?

**Review by aapeliv — COMMENTED** [review]
Yeah, this is what I was thinking in terms of API

File: app/backend/src/couchers/servicers/auth.py (hunk 1 above)
Comment (ShmuelTreiger) [reply app/backend/src/couchers/servicers/auth.py]: Ya, that was one of the bugs which failed

File: app/backend/src/couchers/servicers/auth.py (hunk 2 above)
Comment (ShmuelTreiger) [reply app/backend/src/couchers/servicers/auth.py]: I need to rejigger a few things. Probably reword, just to emphasize it's not called every time

**Pull request merged**